### PR TITLE
feat: update dependencies to use postcss 8

### LINKED ISSUE #36: Upgrade to PostCSS 8
Consider upgrading the plugin to postcss 8 as it offers performance and smaller node_modules size.

### LINKED ISSUE #37: It breaks on “space toggle hack”
See https://css-tricks.com/a-dry-approach-to-color-themes-in-css/ and https://lea.verou.me/2020/10/the-var-space-hack-to-toggle-multiple-values-with-one-custom-property/

Example code:
```css
:root {
  --OFF: ;
}
```

error:
```
TypeError: Cannot read property 'raws' of undefined
    at /Users/filippriabchun/IdeaProjects/ring-ui/src/global/variables.css:2:3
    at Parser.space (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-values-parser/lib/parser.js:387:25)
    at Parser.parseTokens (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-values-parser/lib/parser.js:223:14)
    at Parser.loop (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-values-parser/lib/parser.js:132:12)
    at Parser.parse (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-values-parser/lib/parser.js:51:17)
    at replaceValueSymbols (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-modules-values-replace/index.js:22:60)
    at /Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-modules-values-replace/index.js:193:20
    at /Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss/lib/container.js:55:18
    at Rule.each (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss/lib/container.js:41:16)
    at Rule.walk (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss/lib/container.js:52:17)
    at /Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss/lib/container.js:60:24
    at Root.each (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss/lib/container.js:41:16)
    at Root.walk (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss/lib/container.js:52:17)
    at /Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-modules-values-replace/index.js:190:8
    at async LazyResult.runAsync (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss/lib/lazy-result.js:396:11)
    at async Object.loader (/Users/filippriabchun/IdeaProjects/ring-ui/node_modules/postcss-loader/dist/index.js:97:14)

```

### PR DESCRIPTION
**Pull request by 777PolarFox777**
Changes:
- Updated Postcss to version 8
- Updated code to use new API
- Update other dependencies so they use Postcss 8 as well
- Update tests to work with Postss 8

Fixes https://github.com/princed/postcss-modules-values-replace/issues/36 and fixes https://github.com/princed/postcss-modules-values-replace/issues/37

### CODE REVIEW

**Comment by 777PolarFox777** [comment]
A bit of background here, I faced the same issue as described here https://github.com/princed/postcss-modules-values-replace/issues/37 when I was trying to integrate with Tailwind 3. I did a small research and I've found out that the only thing I need is to update this package to use Postcss 8 so I just did it :D

@princed please let me know if any other changes need to be done

File: package.json [hunk 1]
```diff
@@ -21,17 +21,19 @@
   "dependencies": {
     "enhanced-resolve": "^3.1.0",
     "es6-promisify": "^5.0.0",
-    "icss-utils": "^4.0.0",
+    "icss-utils": "5.1.0",
```
Comment (princed) [inline package.json]: Should probably use caret `^` for consistency, unless there's a reason not to.

**Review by princed — COMMENTED** [review]
Thank you for the PR! ❤️ 

The code looks great to me, would you mind adding the test case you mentioned in #37 too?

File: package.json (hunk 1 above)
Comment (777PolarFox777) [reply package.json]: Done ✅

**Comment by princed** [comment]
Perfect, thank you!

**Comment by princed** [comment]
Published as `4.0.0`. 

Hey friends @WearyMonkey @khanguy00, there's a new major version, it might bring some performance gains (or degradation 😅), but requires update to PostCSS 8 if I'm not mistaken.

**Pull request merged**